subsys/net/lib/conn_mgr/events_handler.c

```diff
@@ -39,10 +39,26 @@ static void conn_mgr_iface_events_handler(struct net_mgmt_event_callback *cb,
 	switch (NET_MGMT_GET_COMMAND(mgmt_event)) {
 	case NET_EVENT_IF_CMD_DOWN:
 		iface_states[idx] &= ~NET_STATE_IFACE_UP;
+
+		if (IS_ENABLED(NET_CONNECTION_MANAGER_AUTO_IF_DOWN) &&
+		    !net_if_is_connectivity_persistent(iface)) {
+			/*
+			 * Take the iface admin-down, if it isn't already.
```

Comment:
I think it should be fine to bring the interface admin down directly from the cb.

Reply:
I thought I had read somewhere that firing a `net_mgmt` event from within the callback of another is a bad idea, but I can't find where I read that, and I'm starting to think I am conflating. It sounds like that's a non-issue, then?

Reply:
Working on the L2 dummy, I stumbled across where I read that warning. It's in `ethernet.h`:

> Carrier ON/OFF handler worker. This is used to create
> network interface UP/DOWN event when ethernet L2 driver
> notices carrier ON/OFF situation. We must not create another
> network management event from inside management handler thus
> we use worker thread to trigger the UP/DOWN event.

It's not actually clear to me **_why_** this is the case, but I'll dig into it.

Reply:
Main reason for this is that the net_mgmt thread stack is quite small and will overflow if we try to do another net mgmt event + other operations from inside the same thread.

Reply:
Thanks for the explanation, @jukkar -- I have delegated the net_if_down call to the conn_mgr thread accordingly